common/lib/transport/connectionmanager.js

```diff
@@ -876,6 +871,15 @@ var ConnectionManager = (function() {
 		this.notifyState({state: 'closed'});
 	};
 
+	ConnectionManager.prototype.onAuthUpdated = function() {
+		var state = this.state.state;
+		if(state == 'connected' || state == 'connecting') {
+			/* in the current protocol version we are not able to update auth params on the fly;
+			 * so disconnect, and the new auth params will be used for subsequent reconnection */
+			this.disconnectAllTransports();
```

Comment:
> Trigger a disconnected event

Well, that's because it is `disconnected`, and isn't `connected` again until the new transport connects.

The way to avoid this is to implement upgrade in all of the other libraries, which I think we agreed would not happen.

This is a temporary situation until we have in-protocol auth update.